### PR TITLE
bug(SUP-481): workflow and resource imports

### PR DESCRIPTION
**Pull request by arturwwl**
### Jira Ticket
SUP-481

### Type
bug

### Title
workflow and resource imports

### CODE REVIEW

File: internal/provider/resources/resource_resource_test.go [hunk 1]
```diff
@@ -176,6 +176,31 @@ func TestResourceResource(t *testing.T) {
 	})
 }
 
+func TestResourceResource_Import(t *testing.T) {
```
Comment (DoronBargo-Etitle-BT) [inline internal/provider/resources/resource_resource_test.go:179]: With ImportStateVerify: false, the tests assert only that the import call doesn't return an error. They don't assert that the imported state matches a fresh Read — which is exactly the round-trip       
  behavior SUP-481 reported broken. 
Right now these tests would pass even if the conversion produced wildly wrong state (e.g., a UUID typo, missing requestable).
                                                                                                                                                                                                                     
I think we need to enable verification and ignore only fields that legitimately can't round-trip.

File: internal/provider/workflows/workflow_data_source.go [hunk 2]
```diff
@@ -549,6 +547,9 @@ func converterWorkflow(
 			}
 
 			if len(rule.ApprovalFlow.Steps) > 0 {
+				ruleModel.ApprovalFlow = &workflowRulesApprovalFlowModel{
```
Comment (DoronBargo-Etitle-BT) [inline internal/provider/workflows/workflow_data_source.go:550]: approval_flow was always present in the data source output (with possibly an empty steps). Post your change, it becomes null when the API returns no steps. Any consumer doing                             
  data.entitle_workflow.x.rules[0].approval_flow.steps will now hit "attempt to access attribute on null value."

File: internal/provider/resources/resource_resource_test.go
Comment (DoronBargo-Etitle-BT) [inline internal/provider/resources/resource_resource_test.go:1]: Please add a test for convertFullResourceResultResponseSchemaToModel with a zero-valued Integration 
  (the synced case)

File: internal/provider/workflows/workflow_data_source.go
Comment (DoronBargo-Etitle-BT) [inline internal/provider/workflows/workflow_data_source.go:1]: Please add a coverage test for the nil-ApprovalFlow path

File: internal/provider/integrations/integration_resource.go
Comment (DoronBargo-Etitle-BT) [inline internal/provider/integrations/integration_resource.go:1]: In integration_resource_test.go please add a test that updates an existing integration's requestable_by_default from true to false and verifies the change persists

File: internal/provider/resources/resource_resource.go [hunk 3]
```diff
@@ -1213,10 +1223,19 @@ func convertFullResourceResultResponseSchemaToModel(
...
 		Workflow:               workflow,
-		Integration: utils.IdNameModel{
-			ID:   utils.TrimmedStringValue(data.Integration.Id.String()),
-			Name: utils.TrimmedStringValue(data.Integration.Name),
-		},
+		// GET /resources/{id} returns "integration": null for synced resources.
+		// Go's JSON decoder zero-values the struct in that case (zero UUID).
+		// Detect this and store nil so the framework can handle the optional field.
+		Integration: func() *utils.IdNameModel {
+			idStr := data.Integration.Id.String()
```
Comment (DoronBargo-Etitle-BT) [inline internal/provider/resources/resource_resource.go:1230]: the id.string always return the the 36-char canonical form — never "". The first half of the OR is dead code, and string-comparing against the zero-UUID literal is hacky.
a better approach will be something like this:
```
      var zeroUUID openapi_types.UUID                                                                                                                                                                                
      if Id == zeroUUID {                                                                                                                                                                                         
          return nil                                                                                                                                                                                                 
      }                                                                                                                                                                                                              
      return &utils.IdNameModel{
          ID:   utils.TrimmedStringValue(id.String()),                                                                                                                                                            
          Name: utils.TrimmedStringValue(Name),                                                                                                                                                                   
      }     
```

File: internal/provider/resources/resource_resource.go [hunk 4]
```diff
@@ -368,6 +370,14 @@ func (r *ResourceResource) Create(ctx context.Context, req resource.CreateReques
 		}
 	}
 
+	if plan.Integration == nil {
+		resp.Diagnostics.AddError(
+			"Client Error",
```
Comment (DoronBargo-Etitle-BT) [inline internal/provider/resources/resource_resource.go:375]: "Client Error" suggests an HTTP/transport problem. This is a config-validation error.
Maybe something like this:
```
if plan.Integration == nil {
      resp.Diagnostics.AddAttributeError(                                                                                                                                                                            
          path.Root("integration"),
          "Missing required attribute",                                                                                                                                                                              
          "`integration` must be set when creating a managed resource. Synced resources cannot be created — they must be imported with `terraform import` or an `import` block.",                                    
      )                                                                                                                                                                                                              
      return                                                                                                                                                                                                         
  } 
```

File: internal/provider/resources/resource_resource.go [hunk 5]
```diff
@@ -368,6 +370,14 @@ func (r *ResourceResource) Create(ctx context.Context, req resource.CreateReques
...
+			"integration is required to create a resource; please specify the integration block with a valid id",
+		)
+		return
+	}
+
 	var integration client.IdParamsSchema
 	integration.Id, err = uuid.Parse(plan.Integration.ID.String())
```
Comment (DoronBargo-Etitle-BT) [inline internal/provider/resources/resource_resource.go:382]: plan.Integration.ID is types.String. Calling .String() on types.String returns the debug representation (e.g., "abc-123-def" with quotes), not the value (abc-123-def). The correct call is  .ValueString().
I think we had this issue in the past in PR https://github.com/entitleio/terraform-provider-entitle/pull/86

**Review by DoronBargo-Etitle-BT — CHANGES_REQUESTED** [review]
Hi, please see my comments

File: internal/provider/integrations/integration_resource.go
Comment (arturwwl) [reply internal/provider/integrations/integration_resource.go:1]: existing test updated to modify this value

File: internal/provider/resources/resource_resource.go (hunk 4 above)
Comment (arturwwl) [reply internal/provider/resources/resource_resource.go:375]: updated

File: internal/provider/resources/resource_resource.go (hunk 5 above)
Comment (arturwwl) [reply internal/provider/resources/resource_resource.go:382]: updated

File: internal/provider/resources/resource_resource.go (hunk 3 above)
Comment (arturwwl) [reply internal/provider/resources/resource_resource.go:1230]: updated

File: internal/provider/workflows/workflow_data_source.go (hunk 2 above)
Comment (arturwwl) [reply internal/provider/workflows/workflow_data_source.go:550]: updated

File: internal/provider/resources/resource_resource_test.go
Comment (arturwwl) [reply internal/provider/resources/resource_resource_test.go:1]: added

File: internal/provider/resources/resource_resource_test.go (hunk 1 above)
Comment (arturwwl) [reply internal/provider/resources/resource_resource_test.go:179]: issue was not about mismatch between config and api. Before fix this test was failing. Issue was that integration was not nullable, so if user did not fill it in .tf it was raising an error. Anyway improved this test a little.

**Review by DoronBargo-Etitle-BT — APPROVED** [review]
lgtm
